spinn_front_end_common/interface/interface_functions/application_runner.py

```diff
@@ -79,12 +58,25 @@ def __call__(
         if notification_interface is not None and wait_on_confirmation:
             notification_interface.wait_for_confirmation()
 
-        # set off the executables that are in sync state
-        if sync_signal is not None:
-            txrx.send_signal(app_id, sync_signal)
-            txrx.wait_for_cores_to_be_in_state(
-                executable_targets.all_core_subsets, app_id,
-                [CPUState.RUNNING, CPUState.PAUSED, CPUState.FINISHED])
+        # set off the executables that are in sync state \
+        # (sending to all is just as safe)
+        if (ExecutableType.USES_SIMULATION_INTERFACE in
+                executable_types.keys() or
+                ExecutableType.SYNC in executable_types.keys()):
+
+            # locate all signals needed to set off executables
+            sync_signals, no_sync_changes = \
+                self._determine_simulation_sync_signals(
+                    executable_types, no_sync_changes)
+
+            # fire all signals as required
+            for sync_signal in sync_signals:
+                txrx.send_signal(app_id, sync_signal)
+
+        # verify all cores are in running states
+        txrx.wait_for_cores_to_be_in_state(
+            executable_targets.all_core_subsets, app_id,
+            [CPUState.RUNNING, CPUState.PAUSED, CPUState.FINISHED])
```

Comment:
Shouldn't this be determined by the executable types?

Reply:
to do that would require massive explosion of the types. you'd need a start, running, end flags separately. was not prepared to go down that road. This works as is. as all cores have to go into those states at some point

Reply:
I think you already have the "end states" don't you?

Reply:
rubbered